components/interop/interop-rest-service-impl/src/main/kotlin/net/corda/interop/rest/impl/v1/InteropRestResourceImpl.kt

```
@@ -46,63 +64,222 @@ internal class InteropRestResourceImpl @Activate constructor(
+        return virtualNodeInfoReadService.getByHoldingIdentityShortHash(ShortHash.of(holdingIdentityShortHash))
+            ?: throw InvalidInputDataException(
+                "No virtual node found with short hash $holdingIdentityShortHash."
             )
-        )
     }
 
     override fun createInterOpIdentity(
-        restInteropIdentity: RestInteropIdentity,
-        holdingidentityshorthash: String
-    ): ResponseEntity<String> {
+        createInteropIdentityRestRequest: CreateInteropIdentityRest.Request,
+        holdingIdentityShortHash: String
+    ): CreateInteropIdentityRest.Response {
+        val vNodeInfo = getAndValidateVirtualNodeInfoByShortHash(holdingIdentityShortHash)
+        val x500Name = MemberX500Name(
+            createInteropIdentityRestRequest.applicationName,
+            vNodeInfo.holdingIdentity.x500Name.locality,
+            vNodeInfo.holdingIdentity.x500Name.country
+        ).toString()
         try {
-            MemberX500Name.parse(restInteropIdentity.x500Name)
+            MemberX500Name.parse(x500Name)
         } catch (e: Exception) {
             throw InvalidInputDataException(
-                "X500 name \"${restInteropIdentity.x500Name}\" could not be parsed. Cause: ${e.message}"
+                "X500 name \"$x500Name\" could not be parsed. Cause: ${e.message}"
             )
         }
+        try {
+            ShortHash.parse(holdingIdentityShortHash)
+        } catch (e: ShortHashException) {
+            throw BadRequestException("Invalid holding identity short hash${e.message?.let { ": $it" }}")
+        }
+        val mapper = ObjectMapper()
+        val result: Map<String, String> = mapper.readValue(
+            createInteropIdentityRestRequest.groupPolicy,
+            object : TypeReference<Map<String, String>>() {})
+        val groupId = interopIdentityWriteService.publishGroupPolicy(
+            result["groupId"].toString(),
+            createInteropIdentityRestRequest.groupPolicy
+        )
```

@julia-filipczak Missing some checks here!
If the group is not "CREATE_ID", or a valid UUID, we need to throw an appropriate error.
Also, if the group ID _is_ "CREATE_ID" we need to throw an error if `members` has any elements.

@jmacmahonr3 @SaurabhR3 Should there also be a check to see if the group is already present, and maybe a comparison with the pre-existing group data?